RCNN_tf.py

```diff
@@ -0,0 +1,338 @@
+        trainable=True,
+        name=name + '_bias'
+    )
+
+    forward_output = tf.nn.conv2d(input_data, forward_weights, [1, 1, 1, 1], padding='SAME')
+
+    output = None
+    for _ in range(depth):
+        recurrent_output = tf.nn.conv2d(cell_states, recurrent_weights, [1, 1, 1, 1], padding='SAME')
+
+        output = tf.add(forward_output, recurrent_output)
+        output += bias
+
+        cell_states = tf.nn.local_response_normalization(
+            tf.maximum(np.asarray(0.0).astype(PRECISION_NP), output),
+            depth_radius=normalization_feature_maps,
+            bias=1,
+            alpha=alpha,
+            beta=beta,
+            name=name + '_lrn'
+        )
+
+    return output
+
+
+def convolutional_layer(
+        input_data,
+        filter_shape,
+        num_input_channels,
+        num_filter,
+        std=.03,
+        stride=(1, 1),
+        padding='same',
+        name='conv'
+):
+    conv_filter_shape = [filter_shape[0], filter_shape[1], num_input_channels, num_filter]
+    strides = [1, stride[0], stride[1], 1]
+
+    weights = tf.Variable(
+        tf.truncated_normal(conv_filter_shape, stddev=std, dtype=PRECISION_TF),
+        trainable=True,
+        name=name + '_weights'
+    )
+
+    if padding.upper() not in PADDING_LIST:
+        raise ValueError('Padding value is not understood')
+
+    output = tf.nn.conv2d(input_data, weights, strides=strides, padding=padding.upper())
+
+    return output
+
+
+def pooling_layer(
+        input_data,
+        pool_shape,
+        stride=(1, 1),
+        padding='same'
+):
+    ksize = [1, pool_shape[0], pool_shape[1], 1]
+    strides = [1, stride[0], stride[1], 1]
+
+    if padding.upper() not in PADDING_LIST:
+        raise ValueError('Padding value is not understood')
+
+    output = tf.nn.max_pool(input_data, ksize=ksize, strides=strides, padding=padding.upper())
+    return output
+
+
+def global_max_pooling_layer(
+        input_data
+):
+    return tf.reduce_max(input_data, axis=[1, 2])
+
+
+def softmax_layer(
+        input_data,
+        num_input_dim,
+        num_output_dim,
+        std=.03,
+        name='softmax'
+):
+    weights = tf.Variable(
+        tf.truncated_normal([num_input_dim, num_output_dim], stddev=std),
+        trainable=True,
+        name=name + '_weights'
+    )
+
+    bias = tf.Variable(
+        tf.truncated_normal([num_output_dim], stddev=std),
+        trainable=True,
+        name=name + '_bias'
+    )
+
+    linear_trans = tf.matmul(input_data, weights) + bias
+    return tf.nn.softmax(linear_trans), linear_trans
+
+
+def accuracy(
+        labels,
+        result
+):
+    correct_prediction = tf.equal(
+        tf.argmax(labels, axis=1),
+        tf.argmax(result, axis=1)
+    )
+
+    return tf.reduce_mean(tf.cast(correct_prediction, PRECISION_TF))
+
+
+if __name__ == '__main__':
+    # Setting the parameters
+    input_shape_ = [None, 28, 28, 1]
+    output_shape_ = [None, 10]
+    learning_rate_ = .1
+    epochs_ = 5
```

Comment:
also more a comment for the future, we should just save checkpoints and make this not so fixed.